modules/gapi/src/backends/ov/govbackend.cpp

```diff
@@ -127,13 +127,23 @@ static int toCV(const ov::element::Type &type) {
 }
 
 static void copyFromOV(const ov::Tensor &tensor, cv::Mat &mat) {
-    const auto total = mat.total() * mat.channels();
-    if (toCV(tensor.get_element_type()) != mat.depth() ||
+    auto total = mat.total() * mat.channels();
+    auto tensor_shape = tensor.get_shape();
+    cv::Mat rsz(mat);
+    if (tensor.get_size() != total) {
+        GAPI_LOG_WARNING(NULL, "copyToOV :: tensor.get_size() != total, resizing cv::Mat");
+        auto W = ov::layout::width_idx("NHWC");
+        auto H = ov::layout::height_idx("NHWC");
```

Comment:
TODO: It might be worth checking the layout.

Reply:
This function is just copy. A dedicated preprocessing routine must be somewhere near (and resize should be already a part of it). @TolyaTalamanov for your consideration

Reply:
https://github.com/opencv/opencv/blob/master/modules/gapi/src/backends/ov/govbackend.cpp#L96 here is it

Reply:
So the real question is... Why the tensor is 1183x490 and the mat is 1180x490? They both are 3-channel.

Reply:
Sure, I keep looking for other option. This is just one of them.
Thanks a lot!